File: backend/src/Models/Document.js
const mongoose = require("mongoose");

const DocumentSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User",
    required: true
  },
  titre: { type: String, required: true },
  description: { type: String, default: "" },
  dateExpiration: { type: Date, required: true },
  notificationEnvoyee: { type: Boolean, default: false },
  dateAjout: { type: Date, default: Date.now }
});

module.exports = mongoose.model("Document", DocumentSchema);

File: frontend/app/src/main/java/ca/uqac/vistudia/Models/Document.kt
package ca.uqac.vistudia.Models

data class DocumentItem(
    val _id: String,
    val titre: String,
    val description: String?,
    val dateExpiration: String,
    val notificationEnvoyee: Boolean = false
)

data class AjouterDocumentRequest(
    val titre: String,
    val description: String,
    val dateExpiration: String
)